packages/itwin/tree-widget/src/test/tree-header/TreeHeader.test.tsx

```diff
@@ -50,37 +50,177 @@ describe("<TreeHeader />", () => {
     });
   });
 
-  it("opens and closes search box", async () => {
-    const { getByRole, container } = render(
+  it("handles buttons overflow when only dropdown button can be seen", async () => {
+    sinon.stub(HTMLElement.prototype, "scrollWidth").get(() => 200);
+    sinon.stub(HTMLElement.prototype, "offsetWidth").get(() => 32);
+
+    const { queryByRole } = render(
       <TreeHeader {...defaultProps}>
         <Button>Button1</Button>
         <Button>Button2</Button>
       </TreeHeader>
     );
 
-    const openButton = getByRole("button", { name: TreeWidget.translate("searchBox.open")  });
-    await userEvents.click(openButton);
-
-    await waitFor(() => expect(container.querySelector(".tree-widget-searchbox.open")).to.not.be.null);
-
-    const closeButton = getByRole("button", { name: TreeWidget.translate("searchBox.close")  });
-    await userEvents.click(closeButton);
-
-    await waitFor(() => expect(container.querySelector(".tree-widget-searchbox.open")).to.be.null);
+    expect(queryByRole("button", { name: "Button1" })).to.be.null;
+    expect(queryByRole("button", { name: "Button2" })).to.be.null;
   });
 
-  it("handles buttons overflow", async () => {
+  it("handles buttons overflow when provided buttons and dropdown button can be seen", async () => {
     sinon.stub(HTMLElement.prototype, "scrollWidth").get(() => 200);
-    sinon.stub(HTMLElement.prototype, "offsetWidth").get(() => 32);
+    sinon.stub(HTMLElement.prototype, "offsetWidth").get(() => 200);
 
     const { queryByRole } = render(
       <TreeHeader {...defaultProps}>
         <Button>Button1</Button>
         <Button>Button2</Button>
+        <Button>Button3</Button>
       </TreeHeader>
     );
 
-    expect(queryByRole("button", { name: "Button1" })).to.be.null;
+    expect(queryByRole("button", { name: "Button1" })).to.not.be.null;
     expect(queryByRole("button", { name: "Button2" })).to.be.null;
+    expect(queryByRole("button", { name: "Button3" })).to.be.null;
+  });
+
+  describe("search box", () => {
+    it("opens and closes search box", async () => {
+      sinon.stub(TreeWidget, "translate").callsFake((key) => key === "searchBox.close" ? "searchBox close button" : "");
+
+      const { getByRole, container } = render(
+        <TreeHeader {...defaultProps} />
+      );
+
+      const openButton = getByRole("button", { name: TreeWidget.translate("searchBox.open")  });
+      await userEvents.click(openButton);
+
+      await waitFor(() => expect(container.querySelector("[aria-label=\"searchBox close button\"]")).to.not.be.null);
```

Comment:
Why not `queryByRole` like used for other button?

Reply:
fixed